lib/carts-abandoned.js

```diff
@@ -46,199 +46,152 @@ const db = new sqlite.Database(process.env.ECOM_AUTH_DB, err => {
-                        db.run('delete from ecomplus_cart_watch where store_id = ? and id = ?', [storeId, cart._id], err => {
-                          if (!err) {
-                            logger.log(`--> cart ${cart._id} deleted / notification Ok /${storeId}`)
-                          }
-                        })
-                      })
-                    requests.push(req)
-                  })
-
-                  return Promise.all(requests)
-                })
-            }
-
-            return getCarts(storeId)
-              .then(carts => {
-                const promises = []
-                if (carts.length) {
-                  for (let i = 0; i < carts.length; i++) {
-                    let createdAt = moment(carts[i].created_at)
-                    let finalDate = moment().diff(createdAt, 'days')
-                    if (finalDate === configObj.is_abandoned_after_days) {
-                      const url = `carts/${carts[i].cart_id}.json`
-                      const promise = appSdk
-                        .apiRequest(storeId, url)
-                        .then(resp => resp.response.data)
-                        .then(cart => {
-                          const { available, completed } = cart
-                          if (available === true && completed === false) {
-                            // send email
-                            return sendMail(cart)
-                          } else if (available === true && completed === true) {
-                            // remove from database
-                            // trow err to pass cart
-                            db.run('delete from ecomplus_cart_watch where store_id = ? and id = ?', [storeId, carts[i].id], err => {
-                              if (!err) {
-                                logger.log(`> cart ${carts[i].id} deleted / status: completed /${storeId}`)
-                              }
-                            })
-                          }
-                        })
-                      promises.push(promise)
-                    }
-                  }
-                }
-                return Promise.all(promises)
-              })
-
-              .then(() => next())
-
-              .catch(e => {
-                logger.error('CartsAbadonedErrors', e)
-                return next()
-              })
-          } else if (err && storeId) {
-            return next()
-          } else if (!next && !err && !storeId && !configObj) {
-            resolve()
-          }
-        })
-      })
-      .catch(e => {
-        logger.error('CartsAbadonedErrors', e)
-      })
-  })
-
-  // restart after 15m
-  const start = () => task().finally(() => setTimeout(() => start(), 5 * 60 * 1000))
-  // init after 30s
-  start()
-}
+const getCarts = () => new Promise((resolve, reject) => {
+  const query = `SELECT * FROM ecomplus_cart_watch WHERE send = ?
+  ORDER BY created_at DESC GROUP BY store_id LIMIT 1600`
```

Comment:
```suggestion
  ORDER BY created_at ASC GROUP BY store_id LIMIT 1600`
```

Deleta os carrinhos velhos demais, depois pega os carrinhos mais velhos (não tão velhos a ponto de serem deletados) primeiro..
Não é isso?

Reply:
Exato a funcao que deleta os carrinho muito velhos (no caso de até 2 meses antes da data atual) 

```
const endDate = new Date()
endDate.setMonth(endDate.getMonth - 2) // Subtract 2 months

const query = `DELETE FROM ecomplus_cart_watch WHERE DATE(created_at) <= DATE(${endDate.toISOString()})`
```